gef-ui/server/controllers/application-details/index.test.js

```diff
@@ -352,27 +350,58 @@ describe('controllers/application-details', () => {
       it('renders `review-decision` when page requested is `review-decision`', async () => {
         mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_APPROVED_WITHOUT_CONDITIONS;
         api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/review-decision'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
 
-        await applicationDetails(MOCKS.MockRequestUrl('/gef/application/123/review-decision'), mockResponse);
+        expect(mockResponse.render)
+          .toHaveBeenCalledWith('partials/review-decision.njk', expect.objectContaining({ applicationStatus: mockApplicationResponse.status }));
+        expect(req.flash).toHaveBeenCalledWith('success');
+      });
+
+      it('renders `unissued-facilities` when page requested is `unissued facilities` ', async () => {
+        mockApplicationResponse.status = CONSTANTS.DEAL_STATUS.UKEF_ACKNOWLEDGED;
+        api.getApplication.mockResolvedValueOnce(mockApplicationResponse);
+        const req = {
+          ...MOCKS.MockRequestUrl('/gef/application/123/unissued-facilities'),
+          flash: jest.fn().mockReturnValue('Facility is updated'),
+        };
+        await applicationDetails(req, mockResponse);
 
         expect(mockResponse.render)
-          .toHaveBeenCalledWith('partials/review-decision.njk', expect.objectContaining({
+          .toHaveBeenCalledWith('partials/unissued-facilities.njk', expect.objectContaining({
             applicationStatus: mockApplicationResponse.status,
+            unissuedFacilitiesPresent: false,
+            facilitiesChangedToIssued: [],
           }));
+        expect(req.flash).toHaveBeenCalledWith('success');
       });
 
-      it('renders `unissued-facilities` when page requested is `unissued facilities` ', async () => {
+      it('to check if flash: jest.fn().mockReturnValue does not exist', async () => {
```

Comment:
This test is still mocking flash -- What we need to do is mock the implimentation of flash when something isn't stored. 

It would be good to call this something a bit less dependant on how our tests are writen, and more what we are trying to test -- for example `passes the success message if success message present in request`

Reply:
Changed to `passes the success message if success message present in request`

Reply:
We can delete this test, as the tests below cover what I'm expecting!